flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/table/MySqlReadableMetadata.java

```diff
@@ -80,6 +80,26 @@ public Object read(SourceRecord record) {
                     return TimestampData.fromEpochMillis(
                             (Long) sourceStruct.get(AbstractSourceInfo.TIMESTAMP_KEY));
                 }
+            }),
+
+    /** Operation type, INSERT/UPDATE/DELETE. */
+    OP_TYPE(
+            "op_type",
+            DataTypes.STRING().notNull(),
+            new MetadataConverter() {
+                private static final long serialVersionUID = 1L;
+
+                @Override
+                public Object read(SourceRecord record) {
+                    final Envelope.Operation op = Envelope.operationFor(record);
+                    if (op == Envelope.Operation.CREATE || op == Envelope.Operation.READ) {
+                        return StringData.fromString(OperationConstants.INSERT);
+                    } else if (op == Envelope.Operation.DELETE) {
+                        return StringData.fromString(OperationConstants.DELETE);
+                    } else {
+                        return StringData.fromString(OperationConstants.UPDATE);
```

Comment:
Maybe we should also consider to distinguish the UPDATE_BEFORE and UPDATE_AFTER.